linkerd/admin/src/main/scala/io/buoyant/linkerd/admin/AdminHandler.scala

```diff
@@ -1,12 +1,11 @@
 package io.buoyant.linkerd.admin
 
-import com.twitter.finagle.http.{MediaType, Request, Response}
-import com.twitter.finagle.{Service, SimpleFilter}
+import com.twitter.finagle.http.{MediaType, Response}
 import com.twitter.util.Future
 import io.buoyant.admin.HtmlView
 import io.buoyant.linkerd.Build
 
-object AdminHandler extends HtmlView {
+class AdminHandler(namerdNav: Boolean) extends HtmlView {
```

Comment:
Interesting, I hadn't really considered that option. 

I agree that changing `AdminHandler` to a class was kind of painful. Currently I don't want to pass `namerdNav` into the `html` method because it means each handler using the function would need to know information about namerd and I'd rather keep handlers self contained. 

Keeping `mkResponse` in an object seems reasonable enough.